src/main/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryProvider.java

```diff
@@ -56,14 +55,35 @@ static InstanceDiscoveryMetadataEntry getMetadataEntry(URL authorityUrl,
                                                            boolean validateAuthority,
                                                            MsalRequest msalRequest,
                                                            ServiceBundle serviceBundle) {
+        String host = authorityUrl.getHost();
+
+        if (shouldUseRegionalEndpoint(msalRequest)) {
+            //Server side telemetry requires the result from region discovery when any part of the region API is used
+            String detectedRegion = discoverRegion(msalRequest, serviceBundle);
+
+            if (msalRequest.application().azureRegion() != null) {
+                host = getRegionalizedHost(authorityUrl.getHost(), msalRequest.application().azureRegion());
```

Comment:
nit: use the `host` variable here for `authorityUrl.getHost()`

Reply:
Good catch, I missed that in a refactoring. Updated in the latest commit to just be 'host'